src/modules/mim/packagemanagerconfiguration.json

```diff
@@ -15,15 +15,8 @@
               {
                 "name": "Packages",
                 "schema": {
-                  "type": "map",
-                  "mapKey":  {
-                    "name": "Name",
-                    "schema": "string"
-                  },
-                  "mapValue": {
-                    "name": "Version",
-                    "schema": "string"
-                  }
+                  "type": "array",
```

Comment:
>"type": "array", [](http://example.com/codeflow?start=18&length=16)

We are still discussing to resolve the problem Magda hit with being unable to remove map elements. That should be supported and we are hoping is just wrong serialization. We should have maps for reported to match desired.

So Ioana I hope will be OK to wait a bit longer on this PR.

Thanks

Reply:
Using an array here is a conscious choice: We need "replace" behavior, not "merge", hence the array. The other discussion thread with Magda is related to the Sample module, I'll come back to that in a separate e-mail.

We could think of using a map in the future if any customer were to ask to have the current package version of all previously installed packages. For that we'd have to implement proper caching to disk to keep track of all packages that were installed in the past. This additional complexity we'd like to avoid at this point in time, because the current customer will essentially always install the same two packages with updated versions. Desired states are captured as DEB packages containing a long list of dependencies using the control-field in the DEB packages' metadata.

Reply:
Just a comment on "We could think of using a map in the future". Once this MIM is done, a PnP interface will be published for it. To make a change a new OSConfig model will have to be published with a new version of the interface. The original interface version would remain. 

And speaking of this, you are planning to prepare a PnP interface for this MIM once the module implementation is complete, yes?

For more details see:  [https://github.com/Azure/azure-osconfig/blob/main/docs/modules.md#14-publishing-dtdl-for-the-module](https://github.com/Azure/azure-osconfig/blob/main/docs/modules.md#14-publishing-dtdl-for-the-module)

Reply:
Thanks for the heads-up! 
Yes, we're planning on publishing the PnP interface once the module has made its way back to upstream.